x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/TokenService.java

```diff
@@ -1178,6 +1178,8 @@ void decryptAndReturnSupersedingTokens(String refreshToken, RefreshTokenStatus r
                     public void onResponse(GetResponse response) {
                         if (response.isExists()) {
                             try {
+                                logger.debug("Found superseding document: index=[{}] id=[{}] primTerm=[{}] seqNo=[{}]",
+                                    response.getIndex(), response.getId(), response.getPrimaryTerm(), response.getSeqNo());
                                 listener.onResponse(
```

Comment:
This is probably paranoid. But I wonder whether it is helpful to have an `assert` here to ensure content from the GetResponse matches data in `refreshTokenStatus`, which is what gets returned to users.

Reply:
Do you mean an assert to verify that `Get` returns the document we asked for? I don't quite follow.

Reply:
What I mean is to compare the token values from the `GET` document and the token values from `superseding.encrypted_tokens`. The latter is what is returned to users here, the former is used for subsequent authentication. It is paranoid because the GetRequest is constructed with `id` using the decrypted access token from `superseding.encrypted_tokens`. So it is kinda like testing the fundamental guarantee of the GET call.